server/src/main/java/io/crate/replication/logical/action/TransportCreateSubscriptionAction.java

```diff
@@ -167,12 +168,14 @@ private static void checkVersionCompatibility(String tableFqn, Version subscribe
         }
     }
 
-    private CompletableFuture<Void> submitClusterStateTask(CreateSubscriptionRequest request,
-                                                           PublicationsStateAction.Response publicationsStateResponse) {
-        var future = new CompletableFuture<Void>();
+    private CompletableFuture<AcknowledgedResponse> submitClusterStateTask(CreateSubscriptionRequest request,
+                                                                           PublicationsStateAction.Response publicationsStateResponse) {
+
+        FutureActionListener<AcknowledgedResponse, AcknowledgedResponse> future = FutureActionListener.newInstance();
+
         clusterService.submitStateUpdateTask(
             source,
-            new ClusterStateUpdateTask() {
+            new AckedClusterStateUpdateTask<>(request, future) {
```

Comment:
Looks like we do not need any new intermediate future/listener, we can just return the `AckedClusterStateUpdateTask.completionFuture()` instead.

Reply:
Removed intermediate future in https://github.com/crate/crate/pull/14926/commits/d55d9e757c1594a0a61e349f00472bebe51767fe